src/core/public/saved_objects/saved_objects_service.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { CoreService } from 'src/core/types';
+import { SavedObjectsClient, SavedObjectsClientContract } from './saved_objects_client';
+
+/**
+ * @public
+ */
+export interface SavedObjectsStart {
+  /** {@link SavedObjectsClient} */
+  client: SavedObjectsClientContract;
+}
+
+export const SavedObjectsService: CoreService<void, SavedObjectsStart> = {
```

Comment:
We recommend using ES6 classes for plugins. Shouldn't core services follow the same convention? 
> Plugins are defined as classes and exposed to the platform itself through a simple wrapper function

https://github.com/restrry/kibana/blob/a1e8858242adcf13dff19fc982c9f3f95424b6a1/src/core/MIGRATION.md#architecture

Reply:
I don't like using classes when there isn't any state anyway. You need to re-type the method return types and create an unnecessary instance. But I changed this to be more consistent with the rest of the code base.